### ISSUE TITLE
Space in ovpn name causing problems with variable lines.

### ISSUE THREAD

**Issue by sheamaus** [haugene/docker-transmission-openvpn#2406]
### Is there a pinned issue for this?

- [X] I have read the pinned issues and could not find my issue

### Is there an existing or similar issue/discussion for this?

- [X] I have searched the existing issues
- [X] I have searched the existing discussions

### Is there any comment in the documentation for this?

- [X] I have read the documentation, especially the FAQ and Troubleshooting parts

### Is this related to a provider?

- [X] I have checked the provider repo for issues
- [X] My issue is NOT related to a provider

### Are you using the latest release?

- [X] I am using the latest release

### Have you tried using the dev branch latest?

- [ ] I have tried using dev branch

### Docker run config used

---
version: '2.1'
services:
 transmission-openvpn:
   volumes:
      - /data/transmission:/config
      - /srv/dev-disk-by-uuid-87ba0797-aab1-48a8-8d4d-094bed0c9245/media/Downloads/BT:/data
      - /etc/localtime:/etc/localtime:ro

   environment:
      - PUID=1000
      - PGID=100
      - CREATE_TUN_DEVICE=true
      - OPENVPN_PROVIDER=IVACY
      - OPENVPN_CONFIG=United Kingdom-London-UDP
      - OPENVPN_USERNAME=<removed>
      - OPENVPN_PASSWORD=<removed>
      - OPENVPN_OPTS=--inactive 3600 --ping 10 --ping-exit 60
      - WEBPROXY_ENABLED=false
      - LOCAL_NETWORK=192.168.5.0/24
      - TRANSMISSION_SCRAPE_PAUSED_TORRENTS_ENABLED=false
      - DNS=8.8.8.8,8.8.4.4

   cap_add:
      - NET_ADMIN

   logging:
      driver: json-file
      options:
         max-size: 10m
   ports:
       - 9091:9091
       - 6789:6789
       - 9696:9696

   restart: always

   image: haugene/transmission-openvpn

 nzbget:
    image: lscr.io/linuxserver/nzbget:latest
    network_mode: "service:transmission-openvpn"
    container_name: nzbget

    environment:
      - PUID=1000
      - PGID=100
      - TZ=Europe/London
      - NZBGET_USER=<removed>
      - NZBGET_PASS=<removed>

    volumes:
      - /data/nzbget:/config
      - /srv/dev-disk-by-uuid-87ba0797-aab1-48a8-8d4d-094bed0c9245/media/Downloads/NZB:/downloads

    restart: unless-stopped

 prowlarr:
    image: lscr.io/linuxserver/prowlarr:develop
    network_mode: "service:transmission-openvpn"
    container_name: prowlarr
    environment:
      - PUID=1000
      - PGID=100
      - TZ=Europe/London
    volumes:
      - /data/prowlarr:/config
    restart: unless-stopped

### Current Behavior

Containers spin up fine, and torrents can be added but can't connect to peers.

### Expected Behavior

Added torrents connect to peers and download.

### How have you tried to solve the problem?

Checking the log I saw the line "/etc/transmission/environment-variables.sh: line 19: export: `Kingdom-Maidenhead-UDP.ovpn': not a valid identifier" and tried a few other UK based servers but none worked. Switched to Belgium-UDP and everything worked as expected.

### Log output

Found configs for IVACY in /tmp/tmp.dM5s3hls1W/vpn-configs-contrib-main/openvpn/ivacy, will replace current content in /etc/openvpn/ivacy
Cleanup: deleting /tmp/tmp.nfYUOigkY5 and /tmp/tmp.dM5s3hls1W
Starting OpenVPN using config United Kingdom-London-UDP.ovpn
Modifying /etc/openvpn/ivacy/United Kingdom-London-UDP.ovpn for best behaviour in this container
Modification: Point auth-user-pass option to the username/password file
Modification: Change ca certificate path
Modification: Change ping options
Modification: Update/set resolv-retry to 15 seconds
Modification: Change tls-crypt keyfile path
Modification: Set output verbosity to 3
Modification: Remap SIGUSR1 signal to SIGTERM, avoid OpenVPN restart loop
Modification: Updating status for config failure detection
Setting OpenVPN credentials...
adding route to local network 192.168.5.0/24 via 172.20.0.1 dev eth0
Mon Nov 14 01:40:21 2022 OpenVPN 2.4.7 x86_64-pc-linux-gnu [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [PKCS11] [MH/PKTINFO] [AEAD] built on Mar 22 2022
Mon Nov 14 01:40:21 2022 library versions: OpenSSL 1.1.1f  31 Mar 2020, LZO 2.10
Mon Nov 14 01:40:21 2022 NOTE: the current --script-security setting may allow this configuration to call user-defined scripts
Mon Nov 14 01:40:21 2022 Outgoing Control Channel Authentication: Using 160 bit message hash 'SHA1' for HMAC authentication
Mon Nov 14 01:40:21 2022 Incoming Control Channel Authentication: Using 160 bit message hash 'SHA1' for HMAC authentication
Mon Nov 14 01:40:21 2022 TCP/UDP: Preserving recently used remote address: [AF_INET]5.254.55.134:53
Mon Nov 14 01:40:21 2022 Socket Buffers: R=[212992->212992] S=[212992->212992]
Mon Nov 14 01:40:21 2022 UDP link local: (not bound)
Mon Nov 14 01:40:21 2022 UDP link remote: [AF_INET]5.254.55.134:53
Mon Nov 14 01:40:22 2022 TLS: Initial packet from [AF_INET]5.254.55.134:53, sid=26f032de bc230a3d
Mon Nov 14 01:40:22 2022 WARNING: this configuration may cache passwords in memory -- use the auth-nocache option to prevent this
Mon Nov 14 01:40:22 2022 VERIFY OK: depth=1, C=HK, ST=Central, L=HK, O=Secure-ServerCA, OU=IT, CN=Secure-ServerCA, name=Secure-ServerCA, emailAddress=mail@host.domain
Mon Nov 14 01:40:22 2022 VERIFY KU OK
Mon Nov 14 01:40:22 2022 Validating certificate extended key usage
Mon Nov 14 01:40:22 2022 ++ Certificate has EKU (str) TLS Web Server Authentication, expects TLS Web Server Authentication
Mon Nov 14 01:40:22 2022 VERIFY EKU OK
Mon Nov 14 01:40:22 2022 VERIFY OK: depth=0, C=HK, ST=Central, L=HK, O=Secure-Server, OU=IT, CN=Secure-Server, name=changeme, emailAddress=mail@host.domain
Mon Nov 14 01:40:22 2022 WARNING: 'link-mtu' is used inconsistently, local='link-mtu 1557', remote='link-mtu 1550'
Mon Nov 14 01:40:22 2022 WARNING: 'cipher' is used inconsistently, local='cipher AES-256-CBC', remote='cipher AES-256-GCM'
Mon Nov 14 01:40:22 2022 WARNING: 'auth' is used inconsistently, local='auth SHA1', remote='auth [null-digest]'
Mon Nov 14 01:40:22 2022 WARNING: 'comp-lzo' is present in remote config but missing in local config, remote='comp-lzo'
Mon Nov 14 01:40:22 2022 Control Channel: TLSv1.2, cipher TLSv1.2 ECDHE-RSA-AES256-GCM-SHA384, 2048 bit RSA
Mon Nov 14 01:40:22 2022 [Secure-Server] Peer Connection Initiated with [AF_INET]5.254.55.134:53
Mon Nov 14 01:40:23 2022 SENT CONTROL [Secure-Server]: 'PUSH_REQUEST' (status=1)
Mon Nov 14 01:40:23 2022 PUSH: Received control message: 'PUSH_REPLY,redirect-gateway def1,dhcp-option DNS 45.74.7.3,dhcp-option DNS 45.74.7.5,sndbuf 393216,rcvbuf 393216,comp-lzo no,route-gateway 45.74.7.161,topology subnet,ping 10,ping-restart 120,ifconfig 45.74.7.168 255.255.255.224,peer-id 6,cipher AES-256-GCM'
Mon Nov 14 01:40:23 2022 OPTIONS IMPORT: timers and/or timeouts modified
Mon Nov 14 01:40:23 2022 OPTIONS IMPORT: compression parms modified
Mon Nov 14 01:40:23 2022 OPTIONS IMPORT: --sndbuf/--rcvbuf options modified
Mon Nov 14 01:40:23 2022 Socket Buffers: R=[212992->425984] S=[212992->425984]
Mon Nov 14 01:40:23 2022 OPTIONS IMPORT: --ifconfig/up options modified
Mon Nov 14 01:40:23 2022 OPTIONS IMPORT: route options modified
Mon Nov 14 01:40:23 2022 OPTIONS IMPORT: route-related options modified
Mon Nov 14 01:40:23 2022 OPTIONS IMPORT: --ip-win32 and/or --dhcp-option options modified
Mon Nov 14 01:40:23 2022 OPTIONS IMPORT: peer-id set
Mon Nov 14 01:40:23 2022 OPTIONS IMPORT: adjusting link_mtu to 1624
Mon Nov 14 01:40:23 2022 OPTIONS IMPORT: data channel crypto options modified
Mon Nov 14 01:40:23 2022 Data Channel: using negotiated cipher 'AES-256-GCM'
Mon Nov 14 01:40:23 2022 Outgoing Data Channel: Cipher 'AES-256-GCM' initialized with 256 bit key
Mon Nov 14 01:40:23 2022 Incoming Data Channel: Cipher 'AES-256-GCM' initialized with 256 bit key
Mon Nov 14 01:40:23 2022 ROUTE_GATEWAY 172.20.0.1/255.255.0.0 IFACE=eth0 HWADDR=02:42:ac:14:00:02
Mon Nov 14 01:40:23 2022 TUN/TAP device tun0 opened
Mon Nov 14 01:40:23 2022 TUN/TAP TX queue length set to 100
Mon Nov 14 01:40:23 2022 /sbin/ip link set dev tun0 up mtu 1500
Mon Nov 14 01:40:23 2022 /sbin/ip addr add dev tun0 45.74.7.168/27 broadcast 45.74.7.191
Mon Nov 14 01:40:23 2022 /sbin/ip route add 5.254.55.134/32 via 172.20.0.1
Mon Nov 14 01:40:23 2022 /sbin/ip route add 0.0.0.0/1 via 45.74.7.161
Mon Nov 14 01:40:23 2022 /sbin/ip route add 128.0.0.0/1 via 45.74.7.161
/etc/transmission/environment-variables.sh: line 19: export: `Kingdom-London-UDP.ovpn': not a valid identifier
sed: can't read /etc/openvpn/ivacy/United: No such file or directory
/etc/transmission/environment-variables.sh: line 19: export: `Kingdom-London-UDP.ovpn': not a valid identifier
Up script executed with device=tun0 ifconfig_local=45.74.7.168
Updating TRANSMISSION_BIND_ADDRESS_IPV4 to the ip of tun0 : 45.74.7.168
TRANSMISSION_HOME is currently set to: /config/transmission-home
Enforcing ownership on transmission config directory
Applying permissions to transmission config directory
Setting owner for transmission paths to 1000:100
Setting permissions for download and incomplete directories

2
Directories: 775
Files: 664
Setting permission for watch directory (775) and its files (664)
-------------------------------------
Transmission will run as
-------------------------------------
User name:   abc
User uid:    1000
User gid:    100
-------------------------------------
Updating Transmission settings.json with values from env variables
Attempting to use existing settings.json for Transmission
Successfully used existing settings.json /config/transmission-home/settings.json
Overriding bind-address-ipv4 because TRANSMISSION_BIND_ADDRESS_IPV4 is set to 45.74.7.168
Overriding download-dir because TRANSMISSION_DOWNLOAD_DIR is set to /data/completed
Overriding incomplete-dir because TRANSMISSION_INCOMPLETE_DIR is set to /data/incomplete
Overriding rpc-password because TRANSMISSION_RPC_PASSWORD is set to [REDACTED]
Overriding rpc-port because TRANSMISSION_RPC_PORT is set to 9091
Overriding rpc-username because TRANSMISSION_RPC_USERNAME is set to 
Overriding scrape-paused-torrents-enabled because TRANSMISSION_SCRAPE_PAUSED_TORRENTS_ENABLED is set to false
Overriding watch-dir because TRANSMISSION_WATCH_DIR is set to /data/watch
sed'ing True to true
STARTING TRANSMISSION
Transmission startup script complete.
/etc/transmission/environment-variables.sh: line 19: export: `Kingdom-London-UDP.ovpn': not a valid identifier
Mon Nov 14 01:40:24 2022 Initialization Sequence Completed

### HW/SW Environment

```markdown
Docker: 5:20.10.21~3-0~debian-bullseye
```

### Anything else?

Seems to me like the space in the config name is causing the issue. I haven't used this in a while so don't know when that started. Also wasn't sure if this would be a problem specific to ivacy or one that spreads to other VPNs so shared it here. Apologies if it's not in the right place.

**Comment by TheRealBenForce** [comment]
I've recently been seeing a "No such file or directory". Rebuilt the container and config volume using latest, but my data volume is on a different host. I don't have a space in my variables. I looked through my docker logs and nothing jumped out at me like what you see. What I've found though is that this error occurs in two ways:

First is when trying to pickup the torrent from the watch folder. The strange part is that the torrent does start downloading and the error goes away after about a minute.
![autoadd](https://user-images.githubusercontent.com/8939575/202081673-873fc2ed-c943-4a26-a4d8-e086f354e116.png)

Second is after about a minute of downloading, the torrent goes into a paused state. Clicking resume will work and clear the error, but again after about a minute the error comes back.
![downloading](https://user-images.githubusercontent.com/8939575/202081912-63820e1f-6a6e-47b3-b87e-c0c0a74d08c0.png)

**Comment by haugene** [comment]
Hey :wave: Seems you're right that there might be some breakage here. #2252 adds "CONFIG" to the persisted environment and when that includes a space it breaks the shell script that is included in many other scripts to persist the environment variables between the openvpn startup and transmission processes.

I've pushed a change to the dev branch. I'm now quoting all the `export` statements and don't think that should cause any trouble with the other variables because any variable in bash is treated as a string anyways(?). If it proves to cause problems we can do a special case and only quote the CONFIG variable.

Try pulling and running the `dev` image and see if that fixes the problem?

**Comment by sheamaus** [comment]
Hey, thanks for the reply and the great work you do on this docker. 

I did notice that the .ovpn configs had changed and I was having some success with them for a while. However I am now finding that torrents aren't able to retrieve metadata or find peers. Checking the settings in Transmission it shows port 51413 as closed. Is it possible these configs are outdated? port 51413:51413 is still correctly listed on the docker compose. I've tried various location configs with them either not able to connect at all like Belgium-Brussels-UDP or connecting and not being able to use port 51413 (Manchester, London, Paris). Thanks

**Comment by sheamaus** [comment]
Hi, not seeing any difference between :latest and :dev probably due to the configs main.zip already having been updated to the .ovpn going back to space-less on there. Both are working the same with the same info appearing in the logs.

Regarding the port there is a setting to port forward in the account area on their website which I have already configures. I have spoken with a live chat agent on their end and they were asking me to test it using their official software and they weren't offering much support regarding the .ovpn configs.

**Comment by sheamaus** [comment]
Just wanted to add that I have resolved my port issue now by adding another Belgium-UDP.ovpn to my own [config fork](https://github.com/sheamaus/vpn-configs-contrib/tree/main/openvpn/ivacy). 
Config was from [here](https://support.ivacy.com/vpnusecases/openvpn-files-windows-routers-ios-linux-and-mac/) using the DD-WRT / Linux / Android / iOS download.
I also tested the United Kingdom-UDP.ovpn from there under "UK-London-UDP.ovpn" in my fork but was finding it coming back with "TLS key negotiation failed". 

Might also be worth noting that through chatting with an agent at IVACY they confirmed that Belgium, Chile, Denmark, Germany, Luxembourg, Netherlands, Russia, Sweden, Turkey are the servers optimised for torrenting.
